src/main/middleware/Authentication.ts

```diff
@@ -0,0 +1,56 @@
+import { NextFunction } from 'express';
+import { Request, Response } from 'express';
+import { auth } from 'firebase-admin/lib/auth';
+import Auth = auth.Auth;
+
+/**
+ * Middleware function to check if the request has correct client header using fireabse-admin auth
+ */
+export class FirebaseAuth {
+  private admin: Auth;
+
+  constructor(firebaseAdmin: Auth) {
+    this.admin = firebaseAdmin;
+  }
+
+  isAuthorizedForNewUser = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<any> => {
+    const idToken: string = req.headers.authorization!;
+
+    try {
+      const decodedToken = await this.admin.verifyIdToken(idToken);
+
+      if (decodedToken) {
+        req.body.idFirebase = decodedToken.uid;
```

Comment:
Minor suggestion- but maybe use `firebaseId` instead of `idFirebase` since you followed `firebaseAdmin` and `firebaseAuth`?

Reply:
Yeah it's better, thanks. I refactored.